### PR TITLE
feat(query): support of Terraform modules for "S3_Bucket_With_Unsecur…

### PR DESCRIPTION
**Pull request by tiagocruz20**
…ed_Cors_Rule"

**Proposed Changes**
- support of Terraform modules for "S3_Bucket_With_Unsecured_Cors_Rule"

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/test/positive2.tf [hunk 1]
```diff
@@ -12,7 +11,7 @@ resource "aws_s3_bucket" "positive2" {
     enabled = false
   }
 
-  cors_rule {
+   cors_rule {
```
Comment (rogeriopeixotocx) [inline assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/test/positive2.tf:14]: Why are you changing this line? Please revert this change.

File: assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/test/positive3.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module "s3_bucket" {
+  source = "terraform-aws-modules/s3-bucket/aws"
+  bucket = "s3-tf-example-versioning"
+  acl    = "private"
+  version = "0.0.1"
+  
+  versioning_inputs = [
```
Comment (rogeriopeixotocx) [inline assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/test/positive3.tf:7]: It seems that `versioning_input`  does not exists for this module, maybe you mean versioning https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/blob/master/variables.tf#L97

File: assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/test/positive4.tf [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module "s3_bucket" {
+  source = "terraform-aws-modules/s3-bucket/aws"
+  bucket = "s3-tf-example-versioning"
+  acl    = "private"
+  version = "0.0.1"
+
+  versioning_inputs = [
```
Comment (rogeriopeixotocx) [inline assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/test/positive4.tf:7]: It seems that `versioning_input`  does not exists for this module, maybe you mean versioning https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/blob/master/variables.tf#L97

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]

File: assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/test/negative2.tf [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      mfa_delete = null
+    },
+  ]
+
+  cors_rule {
+    allowed_methods = ["PUT", "POST"]
+    allowed_origins = ["https://s3-website-test.hashicorp.com"]
+    expose_headers  = ["ETag"]
+    max_age_seconds = 3000
+  }
```
Comment (felipe-avelar) [inline assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/test/negative2.tf:19]: I think this sample is wrong, since `cors_rule` is a list of maps, not a single map, as you can see [here](https://registry.terraform.io/modules/terraform-aws-modules/s3-bucket/aws/latest#input_cors_rule)

File: assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/query.rego [hunk 5]
```diff
@@ -14,5 +14,22 @@ CxPolicy[result] {
...
 		"keyExpectedValue": "'cors_rule' does not allows all methods, all headers or several origins",
 		"keyActualValue": "'cors_rule' allows all methods, all headers or several origins",
+		"searchLine": common_lib.build_search_line(["resource", "aws_s3_bucket", name, "cors_rule"], []),
+	}
+}
+
+CxPolicy[result] {
+	module := input.document[i].module[name]
+	keyToCheck := common_lib.get_module_equivalent_key("aws", module.source, "aws_s3_bucket", "cors_rule")
+	rule := module.cors_rule
```
Comment (felipe-avelar) [inline assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/query.rego:24]: I think here should be an array check, since `cors_rule` is a list of maps, not a single map, as you can see [here](https://registry.terraform.io/modules/terraform-aws-modules/s3-bucket/aws/latest#input_cors_rule). This would look like:
```suggestion
	rule := module.cors_rule[ruleIdx]
```

File: assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/test/positive4.tf [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    },
+  ]
+
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["GET", "PUT", "POST", "DELETE", "HEAD"]
+    allowed_origins = ["*"]
+    expose_headers  = ["ETag"]
+    max_age_seconds = 3000
+  }
```
Comment (felipe-avelar) [inline assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/test/positive4.tf:20]: I think this sample is wrong, since `cors_rule` is a list of maps, not a single map, as you can see [here](https://registry.terraform.io/modules/terraform-aws-modules/s3-bucket/aws/latest#input_cors_rule)

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check comments

File: assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/query.rego [hunk 7]
```diff
@@ -14,5 +14,22 @@ CxPolicy[result] {
...
+	rule := module.cors_rule[ruleIdx]
+	common_lib.unsecured_cors_rule(rule.allowed_methods, rule.allowed_headers, rule.allowed_origins)
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("module[%s].cors_rule", [name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "'cors_rule' does not allows all methods, all headers or several origins",
+		"keyActualValue": "'cors_rule' allows all methods, all headers or several origins",
+		"searchLine": common_lib.build_search_line(["module", name, "cors_rule"], []),
```
Comment (felipe-avelar) [inline assets/queries/terraform/aws/s3_bucket_with_unsecured_cors_rule/query.rego:33]:
```suggestion
		"searchLine": common_lib.build_search_line(["module", name, keyToCheck, ruleIdx], []),
```

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check comments

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Review by rogeriopeixotocx — APPROVED** [review]
